### PR TITLE
Merge PSQLRow into PostgresRow

### PR DESCRIPTION
**Pull request by fabianfett**
Okay, this PR has some meat in it! As it changes the existing PostgresRow quite significantly. We can do this non API breaking, however we may break some users expectations. However we can make those explicit with deprecations.

### Motivation

Currently PostgresRow creates an internal array for every row to allow random access lookup for cells in the row. However I suspect this to not be always necessary. For this reason I propose changing PostgresRow to work like PSQLRow has worked so far. To allow dependent software that depends on the RandomAccessCollection lookups to continue to work correctly we introduce a PostgresRandomAccessRow. This one can be created from a PostgresRow and has all nice O(1) behaviors. 

### Changes

- Add PostgresCell (should be landed extra in #220) before this one
- Remove PSQLRow
- Make PostgresRow light
- Introduce PostgresRandomAccessRow

### Result

- Fastest code by default
- Random access to cells when needed

### CODE REVIEW

**Review by gwynne — APPROVED** [review]
LGTM, just a couple notes

File: Sources/PostgresNIO/Data/PostgresRow.swift [hunk 1]
```diff
@@ -1,77 +1,198 @@
...
+            typeModifier: self.columns[index].dataTypeModifier,
+            formatCode: .binary,
+            value: self.data[column: index]
         )
     }
+}
+
+extension PostgresRow: Equatable {
+    public static func ==(lhs: Self, rhs: Self) -> Bool {
+        lhs.data == rhs.data && lhs.columns == rhs.columns
```
[the remark this answers is not in the document]
Comment (fabianfett) [reply Sources/PostgresNIO/Data/PostgresRow.swift:62]: Yes. Cause it is purely derived from the columns metadata.

**Review by gwynne — APPROVED** [review]
I love cleanups ❤️ More, more! 🙃

**Comment by fabianfett** [comment]
Okay, this was already approved, but I think now it is finally ready... Just to make everyone aware, the important bits are:

1. Cells from `PostgresRow` are not random accessible in O(1) anymore
2. `PostgresRow` now conforms to `Sequence` and `Collection`. The `Element` type is `PostgresCell`. The `Index` type is a custom struct with internal properties. This underlines the `Collection` but not `RandomAccessCollection` properties.
3. `PostgresRandomAccessRow` is a `RandomAccessCollection`. The `Element` type is `PostgresCell`. The `Index` type is `Int`.
4. A PostgresRandomAccessRow is created from a PostgresRow like this:
    ```swift
    var randomAccessRow = row.makeRandomAccess() // create PostgresRandomAccessRow from PostgresRow
   ```

6. The following subscripts are possible.

    ```swift
    let cell = randomAccessRow[0] // access first cell in the row, returns PostgresCell
    let cell = randomAccessRow["id"] // access cell with column name "id" in the row, returns PostgresCell
    let data = randomAccessRow[data: 0] // access first cell in the row, returns PostgresData
    let data = randomAccessRow[data: "id"] // access cell with column name "id" in the row, returns PostgresData
    ```

**Review by gwynne — APPROVED** [review]
Still love it - let's keep it going! 👍 🎉

**Pull request merged**